### PR TITLE
parser: support `admin alter ddl jobs` grammar

### PR DESCRIPTION
**Pull request by fzzf678**
### What problem does this PR solve?

Issue Number: ref #57229

Problem Summary:

### What changed and how does it work?
ref #57229, we need a new grammer to dynamically adjust related parameters during the ddl operation, Similar to the existed `admin cancel ...` and `admin pause ...`, we use `admin alter ...` as an syntax entrance to implement the adjust logic.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/parser/parser.y [hunk 1]
```diff
@@ -11371,6 +11373,33 @@ AdminStmt:
 			Tp: ast.AdminUnsetBDRRole,
 		}
 	}
+|	"ADMIN" "ALTER" "DDL" "JOBS" Int64Num AlterJobOptionList
+	{
+		$$ = &ast.AdminStmt{
+			Tp:        ast.AdminAlterDDLJob,
+			JobNumber: $5.(int64),
```
Comment (fzzf678) [inline pkg/parser/parser.y:11380]: Only one job_id is permitted

File: pkg/parser/parser.y [hunk 2]
```diff
@@ -1093,6 +1093,8 @@ import (
 	AdminStmtLimitOpt                      "Admin show ddl jobs limit option"
 	AllOrPartitionNameList                 "All or partition name list"
 	AlgorithmClause                        "Alter table algorithm"
+	AlterJobOptionList 				       "Alter job option list"
+	AlterJobOption 				           "Alter job option"
```
Comment (tangenta) [inline pkg/parser/parser.y:1097]: Could you run `make fmt` in the parser directory?

Comment (fzzf678) [reply pkg/parser/parser.y]: updated in [c75fe5c](https://github.com/pingcap/tidb/pull/57268/commits/c75fe5c329c3f106c474a2febdb8d2afc2d76c61)

**Review by tangenta — APPROVED** [review]

**Review by CbcWestwolf — APPROVED** [review]

**Review by yudongusa — APPROVED** [review]

**Pull request merged**